specs/_features/eip7732/fork-choice.md

```diff
@@ -623,14 +560,43 @@ def on_payload_attestation_message(
     time_into_slot = (store.time - store.genesis_time) % SECONDS_PER_SLOT
     if is_from_block and time_into_slot >= SECONDS_PER_SLOT // INTERVALS_PER_SLOT:
         return
+```
+
+### `validate_on_attestation`
+
+```python
+def validate_on_attestation(store: Store, attestation: Attestation, is_from_block: bool) -> None:
+    target = attestation.data.target
+
+    # If the given attestation is not from a beacon block message, we have to check the target epoch scope.
+    if not is_from_block:
+        validate_target_epoch_against_current_time(store, attestation)
+
+    # Check that the epoch number and slot number are matching
+    assert target.epoch == compute_epoch_at_slot(attestation.data.slot)
+
+    # Attestation target must be for a known block. If target block is unknown, delay consideration until block is found
+    assert target.root in store.blocks
+
+    # Attestations must be for a known block. If block is unknown, delay consideration until the block is found
+    assert attestation.data.beacon_block_root in store.blocks
+    # Attestations must not be for blocks in the future. If not, the attestation should not be considered
+    slot = store.blocks[attestation.data.beacon_block_root].slot
+    assert slot <= attestation.data.slot
```

Comment:
The comments here are quite long and should be broken up into multiple lines. Note, I haven't re-enable line length limits since adding auto-linting, so nothing is currently complaining. Also inconsistent punctuation.